tools/check-messages-en.js

```diff
@@ -0,0 +1,134 @@
+const fs = require('fs');
+const path = require('path');
+const flowParser = require('flow-parser');
+
+const { parse } = require('recast');
+const { namedTypes: n, visit } = require('ast-types');
+
+const messages_en = require('../static/translations/messages_en.json');
+
+// Make a list of files that might contain UI strings, by recursing in src/.
+const possibleUIStringFilePaths = [];
+const kSrcDirName = 'src/';
+function walk(dir, _dirName = '') {
+  let dirent;
+  // eslint-disable-next-line no-cond-assign
+  while ((dirent = dir.readSync())) {
+    // To reduce false negatives, `continue` when nothing in `dirent` can
+    // cause UI strings to appear in the app.
+
+    if (dirent.isFile()) {
+      // Non-JS code, and Flow type definitions in .js.flow files.
+      if (!dirent.name.endsWith('.js')) {
+        continue;
+      }
+
+      possibleUIStringFilePaths.push(path.join(kSrcDirName, _dirName, dirent.name));
+    } else if (dirent.isDirectory()) {
+      const subdirName = path.join(_dirName, dirent.name);
+
+      // Test code.
+      if (subdirName.endsWith('__tests__')) {
+        continue;
+      }
+
+      walk(fs.opendirSync(path.join(kSrcDirName, subdirName)), subdirName);
+    } else {
+      // Something we don't expect to find under src/, probably containing
+      // no UI strings. (symlinks? fifos, sockets, devices??)
```

Comment:
Yep, this function was inspired by [a function in `tsflower`](https://github.com/gnprice/tsflower/blob/ecb313ccbf75cf4cd0537cb029dc5c9bf1752d7f/src/index.ts#L83-L112). 😛